File: data/EXP01/go/2326/2326.output_3.go
type ListNode struct {
    Val  int
    Next *ListNode
}

func spiralMatrix(m int, n int, head *ListNode) [][]int {
    matrix := make([][]int, m)
    for i := 0; i < m; i++ {
        matrix[i] = make([]int, n)
        for j := 0; j < n; j++ {
            if head != nil {
                matrix[i][j] = head.Val
                head = head.Next
            } else {
                matrix[i][j] = -1
            }
        }
    }
    return matrix
}

}}}}}}}}}}}}}
    m, n := 3, 5
    result := spiralMatrix(m, n, head)
    for i := 0; i < m; i++ {
        fmt.Println(result[i])
    }
}

File: data/EXP01/python/2326/2326.output_3.py
# @lc app=leetcode id=2326 lang=python3
class Solution:
    def spiralMatrix(self, m: int, n: int, head: Optional[ListNode]) -> List[List[int]]:
        matrix = [[-1] * n for _ in range(m)]
        
        directions = [(0, 1), (1, 0), (0, -1), (-1, 0)]
        direction = 0
        
        row, col = 0, 0
        
        while head:
            matrix[row][col] = head.val
            
            dx, dy = directions[direction]
            next_row, next_col = row + dx, col + dy
            
            if next_row < 0 or next_row >= m or next_col < 0 or next_col >= n or matrix[next_row][next_col] != -1:
                direction = (direction + 1) % 4
                dx, dy = directions[direction]
                next_row, next_col = row + dx, col + dy
            
            row, col = next_row, next_col
            head = head.next
        
        return matrix